src/Mutex.php

```diff
@@ -14,18 +14,40 @@ class Mutex
 
   private $_mutexKey;
   private $_unlockOnDestruct;
+  private $_provider;
 
   /**
    * @param IMutexProvider $provider  Cache Pool to store locks
    * @param string         $mutexName The name of the mutex
+   *
+   * @throws \Exception
    */
   public function __construct(IMutexProvider $provider, $mutexName)
   {
+    if(!$this->isValidKey($mutexName))
+    {
+      throw new \Exception('Invalid mutex key');
+    }
+
     $this->_provider = $provider;
     $this->_mutexKey = 'PACKAGED_MUTEX:' . $mutexName;
     $this->_unlockOnDestruct = true;
   }
 
+  /**
+   * @param mixed $key
+   *
+   * @return bool
+   */
+  public function isValidKey($key)
+  {
+    return !(
```

Comment:
I think you should reverse the logic in this instead of just wrapping the whole thing in `!()`, e.g. `return ($key || ($key === 0) || ($key === "0")) && (strlen($key) <= 250)`...